src/coredump/coredump.c

```diff
@@ -597,7 +597,7 @@ static int save_external_coredump(
                 /* tmpfs might get full quickly, so check the available space too.
                  * But don't worry about errors here, failing to access the storage
                  * location will be better logged when writing to it. */
-                if (statvfs("/var/lib/systemd/coredump/", &sv) >= 0)
+                if (fstatvfs(fd, &sv) >= 0)
```

Comment:
mmh but this FD does not point to the directory, it points to the open temp file, is that going to give the same result?

Reply:
The temporary file is created in that directory using `open_tmpfile_linkable`, both in the case when `O_TMPFILE` works and in the case when it doesn't.